x/ecocredit/server/server.go

```diff
@@ -102,7 +102,7 @@ func newServer(storeKey sdk.StoreKey, paramSpace paramtypes.Subspace,
 	}
 	s.sellOrderByBatchDenomIndex, err = orm.NewIndex(sellOrderTableBuilder, SellOrderByBatchDenomIndexPrefix, func(value interface{}) ([]interface{}, error) {
 		denom := value.(*ecocredit.SellOrder).BatchDenom
-		return []interface{}{denom}, nil
+		return []interface{}{[]byte(denom)}, nil
 	}, []byte{})
```

Comment:
we could also just use a string-based index since `denom` is a string

Reply:
When attempting to query using REST, I was getting a panic from the response.

```
panic: unknown field "code" in ecocredit.QuerySellOrdersByBatchDenomResponse
```

Switching to bytes solved the issue. I thought maybe the format of the denom (use of `-`) was causing issues but never got to the bottom of the issue. Any ideas?

Reply:
When attempting to query using REST, I was getting a panic from the response.

```
panic: unknown field "code" in ecocredit.QuerySellOrdersByBatchDenomResponse
```

Switching to bytes solved the issue. I thought maybe the format of the denom (use of `-`) was causing issues but never got to the bottom of it. Any ideas?

Reply:
this is probably because the last parameter was set to `[]byte{}` while you were returning `denom` as a string within the interface slice, those should be of the same underlying type